From 069ab246129be6860aed3389c526543afa87e712 Mon Sep 17 00:00:00 2001
From: Luca BRUNO <luca.bruno@coreos.com>
Date: Thu, 17 Sep 2020 16:07:59 +0000
Subject: [PATCH] vendor/vmw-guestinfo: quickfix to skip performing iopl

This is a quickfix to avoid performing an `iopl`, which is blocked by
kernel_lockdown under SecureBoot.

Refs:
 * https://bugzilla.redhat.com/show_bug.cgi?id=1877995
 * https://github.com/lucab/vmw_backdoor-rs/issues/6
 * https://github.com/coreos/ignition/issues/1092
---
 .../vmware/vmw-guestinfo/vmcheck/vmcheck.go           | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/vendor/github.com/vmware/vmw-guestinfo/vmcheck/vmcheck.go b/vendor/github.com/vmware/vmw-guestinfo/vmcheck/vmcheck.go
index c46cc5e4..ffd866c0 100644
--- a/vendor/github.com/vmware/vmw-guestinfo/vmcheck/vmcheck.go
+++ b/vendor/github.com/vmware/vmw-guestinfo/vmcheck/vmcheck.go
@@ -41,10 +41,13 @@ func IsVirtualWorld() (bool, error) {
 
 // hypervisorPortCheck tests the availability of the HV port.
 func hypervisorPortCheck() (bool, error) {
-	// Privilege level 3 to access all ports above 0x3ff
-	if err := openPortsAccess(); err != nil {
-		return false, err
-	}
+       // XXX(lucab): quickfix for https://github.com/coreos/ignition/issues/1092.
+       /*
+               // Privilege level 3 to access all ports above 0x3ff
+               if err := openPortsAccess(); err != nil {
+                       return false, err
+               }
+       */
 
 	p := &bdoor.BackdoorProto{}
 
-- 
2.21.1

